pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -219,25 +250,81 @@ func (c *Controller) removeFinalizer(pvc *v1.PersistentVolumeClaim) error {
 	return nil
 }
 
-func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, error) {
+// This function checks if a PVC is in use and returns values inUse,
+// needsRequeue and error.
+// isUse: This value indicates if a PVC is in use by any pod or node
+// needsRequeue: This value indicates that the controller loop should
+// requeue the PVC in the controller, instead of waiting for a new event
+// handler call on the PVC. This is an optimization to react faster to a
+// change in state where no pod references the PVC.
+func (c *Controller) isBeingUsed(pvc *v1.PersistentVolumeClaim) (bool, bool, error) {
+	// If PVC is marked for delete, before the PVC.Spec.Volume is populated,
+	// we should not block PVC deletion. By the time PVC is deleted, if PV
+	// creation was already triggered and the volume plugin is in the process
+	// of creation of the volume, then after the PV is created, the PV controller
+	// will trigger a delete for the PV (if it has a reclaim policy of delete),
+	// since the claim will not be found.
+	var pv *v1.PersistentVolume
+	if pvc.Spec.VolumeName != "" {
+		var err error
+		pv, err = c.getPV(pvc)
```

Comment:
what if pv is no longer exist?

Reply:
if PV no longer exists, PV object would be nil, and volume in use by node check will be skipped

Reply:
if pv is nil, currently logic is err != nil, when err != nil, you will return false, false, err which means pvc finalizer will not be removed, right?

Reply:
My bad that is true. Fixing this and will handle a not found case, so that we skip the node check if pv is nil.